From: Shimazu <vanessimbolon2020@gmail.com>
Date: Fri, 31 Jul 2026 02:19:04 +0000
Subject: FireballEffect: Rename variables for maintainability

AE PATCH REASON: Obfuscated variable names hinder maintainability and comprehension.
AE PATCH FIX: Rename var1/var2/var3 to descriptive names throughout the file. Replace teleport() with teleportAsync().
AE PATCH RISK: LOW

---
diff --git a/net/advancedplugins/ae/impl/effects/effects/effects/internal/FireballEffect.java b/net/advancedplugins/ae/impl/effects/effects/effects/internal/FireballEffect.java
--- a/net/advancedplugins/ae/impl/effects/effects/effects/internal/FireballEffect.java	2026-07-30 10:27:31.663350078 +0000
+++ b/net/advancedplugins/ae/impl/effects/effects/effects/internal/FireballEffect.java	2026-07-30 10:28:05.866658428 +0000
@@ -11,34 +11,34 @@
 import org.bukkit.util.Vector;
 
 public class FireballEffect extends AdvancedEffect {
-   public FireballEffect(JavaPlugin var1) {
-      super(var1, "FIREBALL", "Shoot a fireball", "%e");
+   public FireballEffect(JavaPlugin plugin) {
+      super(plugin, "FIREBALL", "Shoot a fireball", "%e");
    }
 
    @Override
-   public boolean executeEffect(ExecutionTask var1, LivingEntity var2, String[] var3) {
-      return this.executeEffect(var1, var2.getLocation(), var3);
+   public boolean executeEffect(ExecutionTask task, LivingEntity entity, String[] args) {
+      return this.executeEffect(task, entity.getLocation(), args);
    }
 
    @Override
-   public boolean executeEffect(ExecutionTask var1, Location var2, String[] var3) {
-      Fireball var4 = (Fireball)var2.getWorld().spawnEntity(var2, EntityType.FIREBALL);
-      var4.setIsIncendiary(false);
-      var4.setYield(0.0F);
-      Vector var5 = new Vector(var2.getX(), var2.getY(), var2.getZ());
-      Vector var6 = new Vector(var4.getLocation().getX(), var4.getLocation().getY(), var4.getLocation().getZ());
-      if (ASManager.isExcessVelocity(var5.subtract(var6))) {
+   public boolean executeEffect(ExecutionTask task, Location location, String[] args) {
+      Fireball fireball = (Fireball)location.getWorld().spawnEntity(location, EntityType.FIREBALL);
+      fireball.setIsIncendiary(false);
+      fireball.setYield(0.0F);
+      Vector target = new Vector(location.getX(), location.getY(), location.getZ());
+      Vector current = new Vector(fireball.getLocation().getX(), fireball.getLocation().getY(), fireball.getLocation().getZ());
+      if (ASManager.isExcessVelocity(target.subtract(current))) {
          return false;
       } else {
-         var4.setVelocity(var5.subtract(var6));
-         var4.teleport(var4.getLocation());
+         fireball.setVelocity(target.subtract(current));
+         fireball.teleportAsync(fireball.getLocation());
 
          try {
-            var4.setVelocity(var5.subtract(var6));
-         } catch (Exception var8) {
+            fireball.setVelocity(target.subtract(current));
+         } catch (Exception e) {
          }
 
-         var4.setYield(0.0F);
+         fireball.setYield(0.0F);
          return true;
       }
    }
